pkg/controller/kibana/controller.go

```diff
@@ -43,12 +64,55 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for updates to the kibana secret
+	secretPred := predicate.Funcs{
+		UpdateFunc:  func(e event.UpdateEvent) bool { return handleSecret(e.MetaNew) },
+		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+	}
+
+	if err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &namespacedMapHandler, secretPred); err != nil {
+		return err
+	}
+
+	// Watch for updates to the global proxy config only
+	proxyPred := predicate.Funcs{
+		UpdateFunc:  func(e event.UpdateEvent) bool { return true },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+		CreateFunc:  func(e event.CreateEvent) bool { return true },
+		GenericFunc: func(e event.GenericEvent) bool { return false },
```

Comment:
as I understand it, `GenericFunc` will trigger when the operator starts up ("does this object already exists?"). Which we don't need -- we just need to know if a proxy object is created or the existing one is updated in order to trigger reconciling Kibana again so it can update itself accordingly.

We don't need the generic since it would be redundant -- the first time we reconcile Kibana if the proxy is out there it will pick this info up.